lxd/devlxd.go

```diff
@@ -297,6 +298,64 @@ func devlxdDevicesGetHandler(d *Daemon, c instance.Instance, w http.ResponseWrit
 	return response.DevLxdResponse(http.StatusOK, c.ExpandedDevices(), "json", c.Type() == instancetype.VM)
 }
 
+var devlxdUbuntuProGet = devLxdHandler{
+	path:        "/1.0/ubuntu-pro",
+	handlerFunc: devlxdUbuntuProGetHandler,
+}
+
+func devlxdUbuntuProGetHandler(d *Daemon, c instance.Instance, w http.ResponseWriter, r *http.Request) response.Response {
+	if shared.IsFalse(c.ExpandedConfig()["security.devlxd"]) {
+		return response.DevLxdErrorResponse(api.NewGenericStatusError(http.StatusForbidden), c.Type() == instancetype.VM)
+	}
+
+	if r.Method != http.MethodGet {
+		return response.DevLxdErrorResponse(api.NewGenericStatusError(http.StatusMethodNotAllowed), c.Type() == instancetype.VM)
+	}
+
+	s := d.State()
+	hostAttachSetting := s.UbuntuPro.GuestAttach()
+	guestAttachSetting := c.ExpandedConfig()["ubuntu_pro.guest_attach"]
+
+	// If either the host or the guest have the value set to "off" (or if the attachment is not configured on the instance), then return "off".
+	if hostAttachSetting == pro.GuestAttachSettingOff || guestAttachSetting == string(pro.GuestAttachSettingOff) || guestAttachSetting == "" {
```

Comment:
As you have access to d.Pro here, can you push this logic down into the ubuntupro package rather than having it in the API handler, then you can add unit tests for it.